### PR TITLE
ROV-166-Auto-fill-filter-options

### PR DESCRIPTION
**Pull request by Vedant-Tyagi**

### CODE REVIEW

**Review by yaneshtyagi — CHANGES_REQUESTED** [review]
Not working as expected.

**Comment by yaneshtyagi** [comment]
@ellipsis review this PR.

File: src/app/application/consolidate-statement/consolidate-statement.component.ts [hunk 1]
```diff
@@ -41,18 +38,31 @@ export class ConsolidateStatementComponent implements OnInit, OnDestroy {
...
+    });
+
+    this.route.queryParams.subscribe(params => {
+      this.statementFiltersForm.patchValue({
+        fromDate: params['fromDate'] ? JSON.parse(params['fromDate']) : null,
+        toDate: params['toDate'] ? JSON.parse(params['toDate']) : null,
+        transactionType: params['transactionType']? JSON.parse(params['toDate']) : null,
+      });
+      this.search();
+      this.fetchStatements();
```
Comment (quy-rov) [inline src/app/application/consolidate-statement/consolidate-statement.component.ts:62]: this is duplicate call to the same function, please avoid.

File: src/app/application/consolidate-statement/consolidate-statement.component.ts [hunk 2]
```diff
@@ -41,18 +38,31 @@ export class ConsolidateStatementComponent implements OnInit, OnDestroy {
...
-      transactionType:[null]
+      account: [null],
+      transactionType: [null]
+    });
+
+    this.route.queryParams.subscribe(params => {
+      this.statementFiltersForm.patchValue({
+        fromDate: params['fromDate'] ? JSON.parse(params['fromDate']) : null,
+        toDate: params['toDate'] ? JSON.parse(params['toDate']) : null,
+        transactionType: params['transactionType']? JSON.parse(params['toDate']) : null,
```
Comment (quy-rov) [inline src/app/application/consolidate-statement/consolidate-statement.component.ts:59]: Why you are mapping `transactionType` to `toDate`?

File: src/app/application/dashboard/dashboard.component.html [hunk 3]
```diff
@@ -1,76 +1,74 @@
...
+        <div class="form-group">
+          <label class="form-label">Transaction Type</label>
+          <ng-select [items]="transactionTypeList" bindLabel="name" bindValue="id" placeholder="Select Transaction Type" formControlName="transactionType">
+          </ng-select>
         </div>
-        <div class="col mt-2">
-            <button class="btn btn-primary btn-green btn-lg" (click)="navigateToConsolidatedStatement()">View</button>
+      </div>
+      <div class="col-3">
+        <div class="form-group">
```
Comment (quy-rov) [inline src/app/application/dashboard/dashboard.component.html:63]: This category filter is not there on Consolidate statement page, how can I map its value?

**Review by quy-rov — CHANGES_REQUESTED** [review]
Changes Required. Majorly, the category filter on dashboard does not have any mapping on the consolidated statement page.

File: src/app/application/dashboard/dashboard.component.html (hunk 3 above)
Comment (yanshityagi) [reply src/app/application/dashboard/dashboard.component.html:63]: @quy-rov Currently we are not mapping the value of the category filter.

File: src/app/application/consolidate-statement/consolidate-statement.component.ts (hunk 1 above)
Comment (Vedant-Tyagi) [reply src/app/application/consolidate-statement/consolidate-statement.component.ts:62]: okay

File: src/app/application/consolidate-statement/consolidate-statement.component.ts (hunk 2 above)
Comment (Vedant-Tyagi) [reply src/app/application/consolidate-statement/consolidate-statement.component.ts]: I have changed the mapping

File: src/app/application/dashboard/dashboard.component.html (hunk 3 above)
Comment (Vedant-Tyagi) [reply src/app/application/dashboard/dashboard.component.html:63]: using query params  and redirecting the page using that

File: src/app/application/dashboard/dashboard.component.ts [hunk 4]
```diff
@@ -53,15 +53,26 @@ export class DashboardComponent implements OnInit {
     this.modalService.open(AddAccountComponent, { centered: true, size: 'md', });
   }
   openConsolidatedStatement() {
-    // Assuming selectedFilters is an object containing selected filter values
     const selectedFilters = {
       fromDate: this.statementFiltersForm.value?.fromDate,
       toDate: this.statementFiltersForm.value?.toDate,
       transactionType: this.statementFiltersForm.value?.transactionType,
       category: this.statementFiltersForm.value?.category
```
[the remark this answers is not in the document]
Comment (Vedant-Tyagi) [reply src/app/application/dashboard/dashboard.component.ts]: It appears that there is no option available on the consolidated statement page. Additionally, I have been instructed not to apply filters if the option is absent.

File: src/app/application/dashboard/dashboard.component.ts [hunk 5]
```diff
@@ -53,15 +53,26 @@ export class DashboardComponent implements OnInit {
     this.modalService.open(AddAccountComponent, { centered: true, size: 'md', });
   }
   openConsolidatedStatement() {
-    // Assuming selectedFilters is an object containing selected filter values
     const selectedFilters = {
```
[the remark this answers is not in the document]
Comment (yanshityagi) [reply src/app/application/dashboard/dashboard.component.ts:53]: @Varuno8 @Vedant-Tyagi If the user does not select any option for Transaction Type on the home screen then the default option will be 'All'. 
So, in this case, if nothing is selected then on the consolidated statement screen the option 'All' should be selected in Transaction Type.

@Vedant-Tyagi Please show "All" as default selected option.

**Pull request closed without merging**